File: app/src/main/java/com/example/customviewapp/BaseActionSheetViewHolder.java
package com.example.customviewapp;

import android.content.Context;
import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

abstract public class BaseActionSheetViewHolder extends RecyclerView.ViewHolder {

    private static final String TAG = BaseActionSheetViewHolder.class.getSimpleName();
    /**Main */
    private Context context;

    public BaseActionSheetViewHolder(Context context, View view) {
        super(view);
        this.context = context;
    }

    public Context getContext() {
        return this.context;
    }

    abstract void bindDataToViewHolder(final ActionSheetViewHolderModel item,final int position);
}

File: app/src/main/java/com/example/customviewapp/BaseActionSheetViewHolder.kt
package com.example.customviewapp

import android.content.Context
import android.view.View
import androidx.recyclerview.widget.RecyclerView

abstract class BaseActionSheetViewHolder : RecyclerView.ViewHolder {

    companion object {
        private val TAG : String = BaseActionSheetViewHolder::class.java.simpleName
    }
    /**Main  */
    private val context : Context

    constructor(context : Context, view : View) : super(view) {
        this.context = context
    }

    public fun getContext() : Context {
        return context
    }

    abstract fun bindDataToViewHolder(item: ActionSheetViewHolderModel, position: Int);
}